src/SqlPersistence/CommandWrapper.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+class CommandWrapper : IDisposable
+{
+    protected DbCommand command;
+
+    public CommandWrapper(DbCommand command)
+    {
+        this.command = command;
+    }
+
+    public DbCommand InnerCommand => command;
+
+    public string CommandText
+    {
+        get { return command.CommandText; }
+        set { command.CommandText = value; }
+    }
+
+    public DbTransaction Transaction
+    {
+        get { return command.Transaction; }
+        set { command.Transaction = value; }
+    }
+
+    public virtual void AddParameter(string name, object value)
+    {
+        var parameter = command.CreateParameter();
```

Comment:
I would love that, but not sure how it could work. This is just the default implementation - OracleCommandWrapper inherits from this and which one you get is determined by the CommandBuilder based on your sqlVariant. The OracleCommandWrapper fills parameters completely differently.

I'm not sure what we could switch on in an extension method, since we don't have a reference to the Oracle components, we can't very well check what kind of command this is, without resorting to GetType() and string comparisons.

OracleCommandWrapper also sets BindByName on the OracleCommand (by reflection, it's gross) on construction.